src/cli/main/cli.py

```diff
@@ -411,11 +412,32 @@ class Config(NamedTuple):
     cache: List[Dict[str, str]]
 
 
+def _get_named_temporary_file_name() -> str:
+    attempts = 0
+    file_name = ""
+    while attempts < 4 and file_name == "":
+        try:
+            with tempfile.NamedTemporaryFile(delete=True) as file:
+                file_name = file.name
+            sleep(1)
+        except FileExistsError as error:
+            CON.print(f"Failed to create {error.filename}, retrying in 1 second...")
+            attempts += 1
+            sleep(1)
+
+    if file_name == "":
```

Comment:
I don't understand the purpose of this. Maybe we should raise an exception if all 5 attempts are failed?

Reply:
The loop tries 4 times, this is the fifth and last and if it still fails then will raise the error

Reply:
Handle it with a boolean so you don't have to repeat the logic outside the while cycle?:

```
success = False

while:
  ...
if not sucess:
  raise exception
```